service/controller/v15/cloudformation/cloud_formation.go

```diff
@@ -56,20 +59,27 @@ func (c *CloudFormation) DescribeOutputsAndStatus(stackName string) ([]*cloudfor
 	// implementations. If the stack creation failed, the outputs can be
 	// unaccessible. This can lead to a stack that cannot be deleted. it can also
 	// be called during creation, while the outputs are still not accessible.
+	var outputsAccessible bool
 	{
-		errorStatuses := []string{
-			cloudformation.StackStatusRollbackInProgress,
+		okStatuses := []string{
+			cloudformation.StackStatusCreateComplete,
 			cloudformation.StackStatusRollbackComplete,
-			cloudformation.StackStatusCreateInProgress,
+			cloudformation.StackStatusUpdateComplete,
+			cloudformation.StackStatusUpdateRollbackComplete,
```

Comment:
Outputs are accessible when stack is in `*Complete` status. We need to consider rollbacks to prevent deadlocks. `StackStatusDeleteComplete` is ignored because at this point or on next reconciliation loop stack should be not found.

Here is a good lecture which confirms that stack should be eventually on one of those states:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-whatis-howdoesitwork.html